articles/140_topic_and_service_name_mapping.md

```diff
@@ -219,13 +219,83 @@ Since ROS 2 topic and service names are expanded to fully qualified names, any b
+Hereby, the partitions field is implemented as an array of strings, where each string is an individual partitions.
+This means further that every partition index is stricly unique and independent from all other items.
+At the same time, DDS Partitions don't support a partition hierarchy.
+If a publisher operates on two partition entries, e.g. `foo` and `bar` with a base name of `baz`, the resulting topic will be twofold:
+
+```
+- /foo/baz
+- /bar/baz
+```
+
+That implies, that DDS Partitions by default are not depicting any hierarchy.
+However, aliasing is easily douable with it.
+That is, having an original topic such as `cmd_vel` in a namespace `robot1`, we can easily duplicate that topic as `robot2/cmd_vel`.
+
+#### Hierarchy with DDS Partitions
+
+Given that DDS Partitions are not hierarchical, we have to setup the hierarchy on top.
+For this, we use the forward slash `/` explicitely in each partition field.
+If we want to replicate a hierarchy of `warehouse`, `robot1`, `camera_left`, and finally the basename `image_raw`, the first entry of the partitions would look like:
+
+```
+warehouse/robot1/camera_left/image_raw
+```
+
+#### Remapping with DDS Partitions
+
+We have to differentiate remapping from aliasing at this point.
+As described before, aliasing implies a duplication a topic.
+Remapping, in contrast, alters an existing topic and thus remaps one topic into another.
+Given the nature of DDS Partitions, aliasing is natively supported by adding a second entry in the partition field.
+Remapping means, we have to change one existing partition field.
+However, as the completely hierarchy is illustrated within the first index of the partitions, the act of remapping is a simple string modification.
+If we want to remap a camera image from `camera1` to `camera2`, we have to modify the string in the respective field.
+Publishers or Subscribers are uneffected by this change and don't have to be shutdown and restarted.
+The way of modifying these strings, whether with complete string replacement or regex modifiers, exceeds the scope of this article.
+
+## Compare and Contrast with ROS 1
+
+In order to support a mapping to the more restrictive DDS topic name rules, these rules are in some ways more constraining than the rules for ROS 1.
+Other changes have been proposed for convenience or to remove a point of confusion that existed in ROS 1.
+In ROS 2, topic and service names differ from ROS 1 in that they:
+
+- must separate the tilde (`~`) from the rest of the name with a forward slash (`/`)
+
+  - This is done to avoid inconsistency with how `~foo` works in filesystem paths versus when used in a ROS name.
+
+- may contain substitutions which are delimited with balanced curly braces (`{}`)
+
+  - This is a more generic extension of the idea behind the tilde (`~`).
+
+- have length limits
+
+  - This is driven by the topic name length limit of DDS.
+
+- may be indicated as "hidden" by using a leading underscore (`_`) in one of the namespaces
```

Comment:
If we support leading underscores do we not need to restrict trailing underscores to avoid ambiguity with the `__` delimiter conversion? Might it not make sense to use a backwards compliant notation such as a trailing underscore to hide things? This seems like it is a change that will break a lot of things when turning on the bridge.